tests/exchange/test_ccxt_compat.py

```diff
@@ -149,6 +165,24 @@ def test_fetch_ohlcv(self, exchange):
         now = datetime.now(timezone.utc) - timedelta(minutes=(timeframe_to_minutes(timeframe) * 2))
         assert exchange.klines(pair_tf).iloc[-1]['date'] >= timeframe_to_prev_date(timeframe, now)
 
+    @pytest.mark.skip("No futures support yet")
+    def test_ccxt_fetch_funding_rate_history(self, exchange_futures):
+        # TODO-lev: enable this test once Futures mode is enabled.
```

Comment:
@samgermain it would be good if you could "play" with this test a bit.

Simply removing the ".mark.skip" on this line, and commenting the "@longrun" at the top of the class will make it easily runnable.

Note: to run this correctly, you'll also need to uncomment the supported modes in each exchange class.
I've tried gateio and ftx (both work), as well as binance (initialization fails, as this is not using dry-run - and "get-brackets" step fails as that requires keys). I'll think of a workaround for this.

The one that concerns me is OKEX - which is completely failing (at least with the latest ccxt version), so maybe you could look at that? (we'll still not be able to enable this test in this PR i think - but it helps us ensure that things do work correctly.

Reply:
So I don't know if this is relevant, but for okex futures (swap on ccxt), the first parameter "symbol" is required, the symbol schema is base/quote:settle (ex. `ZIL/USDT:USDT`, gateio is also like this, and all exchanges will move to that schema sometime soon)

A different method by the name of `fetchFundingHistory`, which we also use, was just merged in a few days ago, so it may not have been in the latest release when you wrote that

Reply:
I don't really know what to do in order to play with it

Reply:
if you look at all changes i did in this file, all (except for one) location will be highlighted.
it's basically:
* remove `@longrun` from the class (this is not highlighed)
* remove `@pytest.mark.skip` from this test
* add `futures: true` to the top of the file for the exchange you wanna test it
* enable "futures, cross" in the exchange class for the exchange to test it on (otherwise you'll get a "does not support cross futures" error)

run `pytest -k test_ccxt_fetch_funding_rate_history`

Reply:
Yeah just update CCXT again, I don't think Okex has a setMarginMode method also, I think it's done similar to Kraken

Reply:
i don't think we'll want to support OKEX at the start anyway ... i think with binance/ftx/ ... we'll have enough to try and handle/test.

Reply:
ftx is only cross, so we won't be able to support it at the start (same with kraken) 
binance, gateio, and okex all provide an isolated  mode

Reply:
i downloaded the newest though and didn't get those error